pkg/integrations/prometheus/promql/parser.go

```diff
@@ -0,0 +1,564 @@
+				Next:            mQuery.MQueryAggs,
+			}
+			mQueryReqs[0].MetricsQuery = mQuery
+		}
+	}
+
+	return mQueryReqs, pqlQuerytype, queryArithmetic, nil
+}
+
+func parsePromQLExprNode(node parser.Node, mQueryReqs []*structs.MetricsQueryRequest, queryArithmetic []*structs.QueryArithmetic,
+	intervalSeconds uint32) ([]*structs.MetricsQueryRequest, []*structs.QueryArithmetic, bool, error) {
+	var err error = nil
+	exit := false
+	mQuery := &mQueryReqs[0].MetricsQuery
+
+	if node == nil {
+		return mQueryReqs, queryArithmetic, exit, nil
+	}
+
+	switch node := node.(type) {
+	case *parser.AggregateExpr:
+		handleAggregateExpr(node, mQuery)
+	case *parser.Call:
+		err = handleCallExpr(node, mQuery)
+	case *parser.VectorSelector:
+		mQueryReqs, err = handleVectorSelector(mQueryReqs, intervalSeconds)
+	case *parser.BinaryExpr:
+		mQueryReqs, queryArithmetic, err = handleBinaryExpr(node, mQueryReqs, queryArithmetic)
+		exit = true
+	case *parser.MatrixSelector:
+		// Process only if the MQueryAggs is nil. Otherwise, it will be handled in the handleCallExpr
+		if mQuery.MQueryAggs == nil {
+			mQueryReqs[0].TimeRange.StartEpochSec = mQueryReqs[0].TimeRange.EndEpochSec - uint32(node.Range.Seconds())
+		}
+	case *parser.SubqueryExpr:
+		// Process only if the MQueryAggs is nil. Otherwise, it will be handled in the handleCallExpr
+		if mQuery.MQueryAggs == nil {
+			mQueryReqs, queryArithmetic, exit, err = parsePromQLExprNode(node.Expr, mQueryReqs, queryArithmetic, intervalSeconds)
+		}
+	case *parser.ParenExpr:
+		// Ignore the ParenExpr, As the Expr inside the ParenExpr will be handled in the next iteration
+	case *parser.NumberLiteral:
+		// Ignore the number literals, As they are handled in the handleCallExpr and BinaryExpr
+	default:
+		log.Errorf("parsePromQLExprNode: Unsupported node type: %T\n", node)
+	}
+
+	if err != nil {
+		log.Errorf("parsePromQLExprNode: Error parsing promql query: %v", err)
+	}
+
+	return mQueryReqs, queryArithmetic, exit, err
+}
+
+func handleAggregateExpr(expr *parser.AggregateExpr, mQuery *structs.MetricsQuery) {
+	aggFunc := expr.Op.String()
+
+	// Handle parameters if necessary
+	if aggFunc == "quantile" && expr.Param != nil {
+		if param, ok := expr.Param.(*parser.NumberLiteral); ok {
+			mQuery.Aggregator.FuncConstant = param.Val
+		}
+	}
+
+	switch aggFunc {
+	case "avg":
+		mQuery.Aggregator.AggregatorFunction = segutils.Avg
+	case "count":
+		mQuery.Aggregator.AggregatorFunction = segutils.Count
+	case "sum":
+		mQuery.Aggregator.AggregatorFunction = segutils.Sum
+	case "max":
+		mQuery.Aggregator.AggregatorFunction = segutils.Max
+	case "min":
+		mQuery.Aggregator.AggregatorFunction = segutils.Min
+	case "quantile":
+		mQuery.Aggregator.AggregatorFunction = segutils.Quantile
+	default:
+		log.Infof("handleAggregateExpr: using avg aggregator by default for AggregateExpr (got %v)", aggFunc)
+		mQuery.Aggregator = structs.Aggregation{AggregatorFunction: segutils.Avg}
```

Comment:
The purpose of this is to use `avg` when no aggregation was provided, right? If so, I think it would be better to move this to a different case (like `case "":`) and then have the `default:` give an error. That way if we get a truly unexpected value like `"hello world"` or a typo, it'll be easier to find the error

Reply:
ohh Yup, I will do that.